### PR TITLE
build openVPN client profile

### PR DESCRIPTION
**Pull request by s2oBCN**
Script to generate an OpenVPN client profile (at least needed in Windows)

This script is to generate the OpenVPN client profile for the ODS in a box.
This profile can be used from an openvpn-connect client, like this https://openvpn.net/downloads/openvpn-connect-v3-windows.msi

### CODE REVIEW

**Comment by stitakis** [comment]
@michaelsauter do we have a formatting standard or/and standard way to build bash scripts? Asking because we have many of them... maybe is time to think about to introduce some conventions for bash scripts.

**Comment by metmajer** [comment]
Thanks, @s2oBCN. Can you please provide some context on how users will benefit from it? You mention Windows, but the paths in the script refer to Unix paths. Some inline documentation on the usage would also help.

**Comment by michaelsauter** [comment]
@stitakis Yes please run them through https://github.com/koalaman/shellcheck. Most more recently added scripts adhere to spellcheck's rules, and are also checked in GitHub Actions, see https://github.com/opendevstack/ods-core/blob/master/.github/workflows/continuous-integration-workflow.yml#L81.

File: ods-devenv/openvpn/build-client-profile.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+#######################################
+# Run this script in the EDP server
```
Comment (metmajer) [inline ods-devenv/openvpn/build-client-profile.sh:4]: Can you rename "in the EDP server" to "in your EDP / ODS box"?

File: ods-devenv/openvpn/build-client-profile.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# ./build-client-profile.sh {EC2_INSTANCE} client1 > clientProfile.ovpn
+# This script is to generate the OpenVPN client profile for the EDP in a box.
+# This profile can be used from an openvpn-connect client
+# (see https://openvpn.net/downloads/openvpn-connect-v3-windows.msi)
+# Arguments:
```
Comment (metmajer) [inline ods-devenv/openvpn/build-client-profile.sh:9]: Can you provide an "Example:" block with an example script call underneath the "Arguments:" block?

File: ods-devenv/openvpn/build-client-profile.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+server=${1?"The server address is required"}
+protocol=udp
+port=1194
+cacert=/etc/openvpn/easy-rsa/keys/ca.crt
+client_cert=/etc/openvpn/easy-rsa/keys/${2?"The client certificate name is required"}.crt
+client_key=/etc/openvpn/easy-rsa/keys/${2}.key
+tls_key=/etc/openvpn/easy-rsa/keys/ta.key
+
+if [ "$#" -lt 2 ]; then
+    echo "params: <server address> <client>"
```
Comment (metmajer) [inline ods-devenv/openvpn/build-client-profile.sh:25]: <client> -> <client-cert-name>?

**Review by metmajer — CHANGES_REQUESTED** [review]
Hi @s2oBCN, a few minor improvements would help. Cheers!

**Comment by metmajer** [comment]
@s2oBCN let's finalise and close down the issue please

**Comment by stitakis** [comment]
@s2oBCN it looks good to me... one check is failing... can you please update the CHANGELOG.md? Just add this feature to the section `## Unreleased`

**Review by stitakis — APPROVED** [review]
looks good to me!

**Review by metmajer — APPROVED** [review]
Like it, @s2oBCN. This comes very handy :) Thanks!

**Pull request merged**